src/modules/site/site.controller.ts

```diff
@@ -36,27 +35,19 @@ export class SiteController {
   @ApiInternalServerErrorResponse({ description: 'An internal server error has occurred.' })
   @ApiBadRequestResponse({ description: 'Bad request.' })
   async getSiteStatusByExporterName(@Query() query: GetSiteStatusByExporterNameQueryDto, @Res() res: Response): Promise<void> {
-    try {
-      const getSiteStatusByExporterNameResponse = await this.service.getSiteStatusByExporterName(query.exporterName);
-      if (getSiteStatusByExporterNameResponse.status === 'Failed') {
-        res.status(HttpStatusCode.FailedDependency).json(getSiteStatusByExporterNameResponse);
-        return;
-      }
-      if (getSiteStatusByExporterNameResponse.status === 'Created') {
-        res.status(HttpStatusCode.Ok).json(getSiteStatusByExporterNameResponse);
-        return;
-      }
-      if (getSiteStatusByExporterNameResponse.status === 'Provisioning') {
-        res.status(HttpStatusCode.Accepted).json(getSiteStatusByExporterNameResponse);
-        return;
-      }
-      throw new InternalServerErrorException(`Received unexpected status "${getSiteStatusByExporterNameResponse.status}"`);
-    } catch (error) {
-      if (error instanceof SiteNotFoundException) {
-        res.status(HttpStatusCode.NotFound).json({});
-        return;
-      }
-      throw error;
+    const getSiteStatusByExporterNameResponse = await this.service.getSiteStatusByExporterName(query.exporterName);
+    if (getSiteStatusByExporterNameResponse.status === 'Failed') {
+      res.status(HttpStatusCode.FailedDependency).json(getSiteStatusByExporterNameResponse);
+      return;
     }
+    if (getSiteStatusByExporterNameResponse.status === 'Created') {
+      res.status(HttpStatusCode.Ok).json(getSiteStatusByExporterNameResponse);
+      return;
+    }
+    if (getSiteStatusByExporterNameResponse.status === 'Provisioning') {
```

Comment:
Can we use the enum instead of referring to the strings directly here please? (I think it was just missed when you added the enum)